### PR TITLE
Make linear swap percentage change stricter

### PR DESCRIPTION
**Pull request by nventuro**
Follow up to #1028. The underlying issue is similar to the one fixed in #998.

### CODE REVIEW

File: pkg/pool-linear/contracts/LinearPool.sol [hunk 1]
```diff
@@ -605,19 +601,34 @@ contract LinearPool is BasePool, IGeneralPool, LinearMath, IRateProvider {
...
+        (uint256 lowerTarget, uint256 upperTarget) = getTargets();
+        _require(_isMainBalanceWithinTargets(lowerTarget, upperTarget), Errors.OUT_OF_TARGET_RANGE);
+
+        super.setSwapFeePercentage(swapFeePercentage);
+    }
+
+    function _isMainBalanceWithinTargets(uint256 lowerTarget, uint256 upperTarget) private view returns (bool) {
+        bytes32 poolId = getPoolId();
+        (, uint256[] memory balances, ) = getVault().getPoolTokens(poolId);
+        uint256 mainTokenBalance = _upscale(balances[_mainIndex], _scalingFactor(_mainToken));
```
[the remark this answers is not in the document]
Comment (EndymionJkb) [reply pkg/pool-linear/contracts/LinearPool.sol:627]: Also could test the UNHANDLED_BY_LINEAR_POOL exit (e.g., try to exit when unpaused with a non-proportional exit kind).

**Review by EndymionJkb — CHANGES_REQUESTED** [review]

File: pkg/pool-linear/contracts/LinearPool.sol [hunk 2]
```diff
@@ -605,19 +601,34 @@ contract LinearPool is BasePool, IGeneralPool, LinearMath, IRateProvider {
...
+
+        return mainTokenBalance >= lowerTarget && mainTokenBalance <= upperTarget;
```
Comment (dmf7z) [inline pkg/pool-linear/contracts/LinearPool.sol:629]: Range is inclusive [lowerTarget, upperTarget] so signs are ok.

**Review by dmf7z — APPROVED** [review]

File: pkg/pool-linear/contracts/LinearPool.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (nventuro) [reply pkg/pool-linear/contracts/LinearPool.sol:627]: Sorry, I don't follow. What OUT_OF_BOUNDS?

re. the gas savings, we're not currently struggling for bytecode, and your change might be misleading. That property only holds for main token: for the wrapped token `_scalingFactor()` does not equal the state variable - I don't want to make it seems like it might.

[the remark this answers is not in the document]
Comment (EndymionJkb) [reply pkg/pool-linear/contracts/LinearPool.sol:627]: For the OUT_OF_BOUNDS, there's an edge case code path that's not triggered by explicit tests. Like the test itself, not really necessary, but easy to add.

```
        // Sanity check: this is not entirely necessary as the Vault's interface enforces the indices to be valid, but
        // the check is cheap to perform.
        _require(indexIn < _TOTAL_TOKENS && indexOut < _TOTAL_TOKENS, Errors.OUT_OF_BOUNDS);
```
